src/main/java/com/snowflake/kafka/connector/config/KafkaProviderValidator.java

```diff
@@ -0,0 +1,33 @@
+package com.snowflake.kafka.connector.config;
+
+import static com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig.PROVIDER_CONFIG;
+
+import com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
+
+/** Validator to validate Kafka Provider values which says where kafka is hosted */
+class KafkaProviderValidator implements ConfigDef.Validator {
+  public KafkaProviderValidator() {}
+
+  // This API is called by framework to ensure the validity when connector is started or when a
+  // validate REST API is called
+  @Override
+  public void ensureValid(String name, Object value) {
+    assert value instanceof String;
+    final String strValue = (String) value;
```

Comment:
should we assert on instance of string?

Reply:
It's copied. Even if it's wrong I don't want to change anything within this PR.